---
title: "Final Report: Group 2"
author: "Jacob Gutierrez, Patty Langasek, Jennifer Sullivan"
date: 12/13/18
output: 
 html_document:
    df_print: paged
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval=TRUE)
```


##Libraries: 
```{r libraries, message=FALSE}
library(tidyverse)
library(skimr)
library(here)
library(knitr)
```


## Paper:
### Common Genetic Variants in _FOXP2_ Are Not Associated with Individual Differences in Language Development
Authors: Katheryn L. Mueller, Jeffrey C. Murray, Jacob J. Michaelson, Morten H. Christiansen, Sheena Reilly, J. Bruce Tomblin

Published open access through PLOS One on April 11, 2016.

[Paper can be found here](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0152576)
[All Data can be found here](https://zenodo.org/record/44640#.XBLLqMtKiV4)

## Paper Summary:
Mueller et. al., asked the question: Can individual differences in language ability normally seen in the general population be partially explained by any of the 13 common SNPs in the _FOXP2_ gene? Histocial context implicates the _FOXP2_ gene as having an effect on language development. Previous studies have evaluated single nucleotide polymorphism (SNP) differences in small family-based comparisons with siblings displaying variance in language abilities, but prior to Mueller et. al, no studies had evaluated language differences and _FOXP2_ SNP variances that can be genearlized to the population.

As part of their analysis, the authors used data collected from, ultimately, 3 previous studies that evaluated language scores (at minimum) and collected DNA samples from the study participants:

The “longitudinal” study assessed the prevalence of language disorders in 7,218 kindergarten-age children, and the subsequent outcomes for those with language impairment and without.  This initial study intentionally oversampled for children with language impairment.  Thus, in order to select a sample that is more representative of the general population, the current study assigned a weight value to each subject equal to the reciprocal of the probability of selecting that subject.  A subset of 500 subjects was then sampled, with higher-weighted subjects more likely to be selected than lower-weighted subjects.  All subjects spoke English and had normal hearing and no reported neurodevelopmental disabilities.  Only Caucasian subjects were selected.

Language ability was indicated by a composite score based on five subtests of the Test of Language Development-2:P.  The tests were chosen to evaluate the subjects’ abilities in the following hildren with more normal scores.  All subjects had normal hearing and no reported neareas: vocabulary, grammar, narration, comprehension, and expression.  The Block Design and Picture Completion tests of the Wechsler Preschool and Primary Scale of Intelligence were used to identify and remove any subject with a nonverbal IQ under 70 in order to prevent confounding by anyone with intellectual disability.

The “school” study assessed language abilities in first- through fourth-grade children.  318 subjects were recruited from this study based on their performance on the verbal subscales of the Iowa Tests of Basic Skills.  Children with low language scores were selected, along with a random sample of curodevelopmental disabilities.  Only Caucasian subjects were selected.

Language ability was indicated by a composite score based on a series of tests similar to that administered to the “longitudinal” cohort, but adjusted to better reflect the skill levels of the different age groups in the “school” cohort.  The Block Design and Picture Completion tests of the Wechsler Preschool and Primary Scale of Intelligence were used to identify and remove any subject with a nonverbal IQ under 70 in order to prevent confounding by anyone with intellectual disability.

The ELVS (Early Language in Victoria) dataset was consulted to control for an association between poor language abilities and overall language ability due to concerns of the variance between the longitudinal and school studies. It was a longitudinal birth cohort study from Australia that, using a portion of the same assessments as the Iowa longitudinal and school samples (sentence structure, concepts & direction, recalling sentences, and word structure), studied language ability of a sample of the population. Each subject is on its own line, however, only one SNP (rs1916988) is considered for each subject. The dataset is the language score, not the standardized z-score from the Iowa-based studies.

Mueller et. al. found no significant indication that the common _FOXP2_ SNPs found in the general population are associated with population-level language ability differences.

# Exploratory Data Analysis Report

## Discussion: data quality issues
Though the data was freely and readily available, there was no explanation of what the data was or what it meant. An accompanyhing codebook would have helped navigate the ambiguity of raw scores vs. z-scores between the two datasets, as well as their notation for missing allele (0).

Presenation of the data in a more visual format instead of tables of text would have also helped better appreciate what the raw data was and what was being evaluated.

The authors also didn't discuss how exactly they curated the data for analysis. As we moved forward with replication, our counts did not match theirs, and we never did find the discrepancy.


## Stats and plots


Please Ensure that the foxp2_tidy before importing to ensure generation of tidy data. 
```{r read_in, message = FALSE}
elvs <- read_csv(here::here("data", "tidy_elvs.csv"))
elvs <- elvs %>% mutate(rs1916988 = as_factor(rs1916988))
iowa <- read_csv(here::here("data", "tidy_iowa.csv"))
iowa <- iowa %>% mutate_at(vars(starts_with("rs")), funs( as_factor(.)))
```
Since these the data comes in two seperate files they will be assessed seperately. 

Additionally since there is no easy way to keep factor levels stored in csv files the recode statements are below:
```{r recode}
# C start
CC <- 'C/C'
CG <- 'C/G'
CT <- 'C/T'

#G start
GG <- 'G/G'
GA <- 'G/A'
GT <- 'G/T'

#T start
TT <-  'T/T'
TC <- 'T/C'


#A start
AA <- 'A/A'
AC <- 'A/C'
AG <- 'A/G'

#iowa %>% count(rs11505922) %>% glimpse() #used to formulate levels at each variant

# rs7791396
iowa  <-  iowa %>% mutate(rs7791396 = factor(rs7791396, levels = c(CC,CG,GG)))
# rs12155328
iowa <-  iowa %>% mutate(rs12155328 = factor(rs12155328, levels = c(CC,CT,TT)))
# rs10447760
iowa <-  iowa %>% mutate(rs10447760 = factor(rs10447760, levels = c(CC,CT,TT)))
# rs10953754
iowa <-  iowa %>% mutate(rs10953754 = factor(rs10953754, levels = c(AA,AG,GG)))
# rs2244419
iowa <-  iowa %>% mutate(rs2244419 = factor(rs2244419, levels = c(CC,CT,TT)))
# rs1668335
iowa <-  iowa %>% mutate(rs1668335 = factor(rs1668335, levels = c(AA,GA,GG)))
# rs2396720
iowa <-  iowa %>% mutate(rs2396720 = factor(rs2396720, levels = c(AA,AC,CC)))
# rrs1916988
iowa <-  iowa %>% mutate(rs1916988 = factor(rs1916988, levels = c(CC,TC,TT)))
# rs11505992
iowa <-  iowa %>% mutate(rs11505922 = factor(rs11505922, levels = c(CC,CT,TT)))
# rs7785701
iowa <-  iowa %>% mutate(rs7785701 = factor(rs7785701, levels = c(CC,CG,GG)))
# rs2106900
iowa <-  iowa %>% mutate(rs2106900 = factor(rs2106900, levels = c(AA,GA,GG)))
# rs7799652
iowa <-  iowa %>% mutate(rs7799652 = factor(rs7799652, levels = c(GG,GT,TT)))
# rs1005958
iowa <-  iowa %>% mutate(rs1005958 = factor(rs1005958, levels = c(CC,TC,TT)))


# ELVS dataset:
# rrs1916988
elvs <-  elvs %>% mutate(rs1916988 = factor(rs1916988, levels = c(CC,CT,TT)))

```

## ELVS dataset: 

###Raw data 
```{r elv_glimp}
elvs %>% glimpse()
```

number of variables: 3 (including the index we created (X1))  
number of observations: 308  
numerical variable: CELF7SS0 (the language score)  
categorical variable: rs1916988 (the SNP in the FOXP2 gene)

Each observation represents a research participant (a child around seven years of age).


###Compute summary statistics.
```{r}
elvs %>% skim()
```


A few things to note:  
1.  There are 308 observations, with no missing values.  (Originally there were 328 observations, but we removed 20 observations that had no genotype information.)  
2.  The three genotypes for the rs1916988 SNP are C/T, T/T, and C/C, with C/T being the most common (47%), closely followed by T/T (43%), and C/C being the least common (10%).  
3.  According to the paper, the language scores were standardized to have a mean of 100 and a standard deviation of 15.  The true mean and standard deviation calculated from this data are close to the stated values: 99.04 and 14.08, respectively.  
4.  The language scores range from 45 to 134.


### Distribution of the CELF7SS0 language scores

```{r celf}
elvs %>% ggplot(aes(x=CELF7SS0, fill = "red")) + geom_density() + geom_vline(xintercept = 100)
```

Here we can see that the distribution of the CELF7SS0 language score is centered at approximately 100. We also notice that it is left-skewed, indicating that we have a larger proportion of low scores present in this data set.  Therefore, it appears that children with weaker language skills were oversampled in this data set.  This study was conducted to find correlations between genotype and normal differences in language ability seen in the general population.  However, this higher prevalence of low scores suggests that this sample may not be entirely representative of the general population.

#### The frequency of each genotype for the rs1916988 SNP

```{r}
elvs %>% ggplot(aes(x=rs1916988,fill=rs1916988)) + geom_bar() 
```

Here we can see that the C/T genotype was the most common variant in the data set, with a frequency just a bit higher than the T/T genotype. Additionally there was a very small proportion of participants with the C/C genotype (just 10% of participants had this variant).


#### The distribution of the CELF7SS0 language scores, colored by genotype for rs1916988

```{r}
elvs %>% ggplot(aes(x=CELF7SS0, fill = rs1916988, alpha = 0.7)) + 
  geom_density() + 
  labs(title = "Language Scores from ELVS dataset, colored by Genotype variant for rs1916988")
```

This plot shows that the lower language scores are associated with the presence of the C allele.  This is consistent with what is reported in the article, though the authors state that the differences in the mean language scores for the three genotypes were not significant.  We can see that much of the data for all three variants does fall within a similar range of scores.

## IOWA dataset: 

###Raw data:

```{r}
iowa %>% glimpse()
```

number of variables: 17 (including the index we created (X1) and the ID that was assigned to each research participant)  
number of observations: 834
numerical variable: LCOMP (the language score)
categorical variables: Sample (the study from which the subject was sampled), each of 13 SNPs in the FOXP2 gene

Each observation represents a research participant (a kindergarten-age child from the Longitudinal study or an elementary school-age (first- through fourth-grade) child from the School study).

### Summary statistics:

```{r}
iowa %>% skim() 
```

A few things to note:  
1.  There are 834 observations, with no missing values.  However, all of the SNPs contain a significant proportion of entries labeled as "0/0."  It is presently unclear if these represent missing data or a homozygous deletion.  
2.  The three genotypes for the rs1916988 SNP (examined in the ELVS data set) are T/C, T/T, and C/C, with T/C being the most common (44%), followed by T/T (42%) and C/C being the least common (9%).  Possible deletion accounted for just 4%.
3.  The data was taken as a subset of data from two larger studies (Longitudinal and School).  The original samples of language scores were normalized to have a mean of 0 and a standard deviation of 1.  However, the mean for the selected subset of participants is -0.35, indicating that children with lower language ability were oversampled.  The standard deviation is slightly higher than that for the original data sets at 1.09.  
4.  The language scores range from -3.45 to 2.59.

#### Distribution of the LCOMP language scores, colored by the sample origin

```{r}
iowa %>% ggplot(aes(x=LCOMP,fill=Sample,alpha=.9)) + 
  geom_density()
```
The School sample is fairly symmetrical and centered approximately at 0, suggesting it is likely a good representation of the general population.  However, we can see that the Longitudinal sample is bimodal, with the taller peak (and therefore the majority of the language scores) below 0.  This is a clear illustration that the Longitudinal sample has an overabundance of children with lower-than-average language abilities (in spite of the authors' effort to correct for this bias when selecting their sample).  Thus, the Longitudinal sample does not appear to represent the general population.

#### The frequency of each genotype for the rs1916988 SNP

```{r}
iowa %>% ggplot(aes(x=rs1916988,fill = rs1916988, alpha = .9)) + 
  geom_bar(position = "stack") 
```
When comparing the same SNP in both the IOWA data set and the ELVS data set, we can see that there is a similar distribution of genotypes bewtween both samples. This indicates that at least both studies were designed similarly and sampled from similar populations.

```{r common_SNP}
common_SNP <- iowa %>% select(LCOMP, Sample, rs1916988)

common_SNP %>% 
  ggplot(aes(x = LCOMP, color = rs1916988, alpha = .7)) + geom_density() + facet_wrap(~Sample) + ggtitle('Iowa Score for rs1916988 by sample')
```

The reason the authors decided to look at the rs1916988 SNP in particular is because they got opposing results for this SNP when they looked at the Longitudinal and School samples separately.  The authors state that in the School sample, the T/T genotype was associated with significantly lower language scores than the T/C genotype.  In the Longitudinal sample, they got the opposite result, with the T/T genotype associated with higher language scores than either the T/C or C/C genotype, though this result was not significant.  From these plots, it is not apparent that these differences between genotypes would exist, as the distributions for either cohort seem to be quite similar for all genotypes.


## Comments



# Replication Report:

## Table1:


## Table2:

## Table3:


## Table4:



# Extension Report:

## Question:

 explanation about SNP effects interacte in combination!
 
## Plot:




## ANCOVA:



# Summary:


 


