---
title: "Final Report: Group 2"
author: "Jacob Gutierrez, Patty Langasek, Jennifer Sullivan"
date: 12/13/18
output: 
 html_document:
    df_print: paged
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval=TRUE)
```


##Libraries: 
```{r libraries, message=FALSE}
library(tidyverse)
library(skimr)
library(here)
library(knitr)
library(moderndive)
library(psycho)
library(BBmisc)
library(car)
library(phia)
```


## Paper:
### Common Genetic Variants in _FOXP2_ Are Not Associated with Individual Differences in Language Development
Authors: Katheryn L. Mueller, Jeffrey C. Murray, Jacob J. Michaelson, Morten H. Christiansen, Sheena Reilly, J. Bruce Tomblin

Published open access through PLOS One on April 11, 2016.

[Paper can be found here](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0152576)
[All Data can be found here](https://zenodo.org/record/44640#.XBLLqMtKiV4)

## Paper Summary:
Mueller et. al., asked the question: Can individual differences in language ability normally seen in the general population be partially explained by any of the 13 common SNPs in the _FOXP2_ gene? Histocial context implicates the _FOXP2_ gene as having an effect on language development. Previous studies have evaluated single nucleotide polymorphism (SNP) differences in small family-based comparisons with siblings displaying variance in language abilities, but prior to Mueller et. al, no studies had evaluated language differences and _FOXP2_ SNP variances that can be generalized to the population.

As part of their analysis, the authors used data ultimately collected from 3 previous studies that evaluated language scores (at minimum) and collected DNA samples from the study participants.

The “longitudinal” study assessed the prevalence of language disorders in 7,218 kindergarten-age children, and the subsequent outcomes for those with language impairment and without.  This initial study intentionally oversampled for children with language impairment.  Thus, in order to select a sample that is more representative of the general population, Mueller et al. assigned a weight value to each subject equal to the reciprocal of the probability of selecting that subject.  A subset of 500 subjects was then sampled, with higher-weighted subjects more likely to be selected than lower-weighted subjects.  All subjects spoke English and had normal hearing and no reported neurodevelopmental disabilities.  Only Caucasian subjects were selected.

Language ability was indicated by a composite score based on five subtests of the Test of Language Development-2:P.  The tests were chosen to evaluate the subjects’ abilities and included children with more normal scores.  All subjects had normal hearing and no reported issues in the areas of vocabulary, grammar, narration, comprehension, and expression.  The Block Design and Picture Completion tests of the Wechsler Preschool and Primary Scale of Intelligence were used to identify and remove any subject with a nonverbal IQ under 70 in order to prevent confounding by anyone with intellectual disability.

The “school” study assessed language abilities in first- through fourth-grade children.  318 subjects were recruited from this study based on their performance on the verbal subscales of the Iowa Tests of Basic Skills.  Children with low language scores were selected, along with a random sample of curodevelopmental disabilities.  Only Caucasian subjects were selected.

Language ability was indicated by a composite score based on a series of tests similar to that administered to the “longitudinal” cohort, but adjusted to better reflect the skill levels of the different age groups in the “school” cohort.  The Block Design and Picture Completion tests of the Wechsler Preschool and Primary Scale of Intelligence were used to identify and remove any subject with a nonverbal IQ under 70 in order to prevent confounding by anyone with intellectual disability.

The ELVS (Early Language in Victoria) dataset was consulted to control for an association between poor language abilities and overall language ability due to concerns of the variance between the longitudinal and school studies. It was a longitudinal birth cohort study from Australia that, using a portion of the same assessments as the Iowa longitudinal and school samples (sentence structure, concepts & direction, recalling sentences, and word structure), studied language ability of a sample of the population. Each subject is on its own line, however, only one SNP (rs1916988) is considered for each subject. The dataset is the language score, not the standardized z-score as in the Iowa-based studies.

Mueller et. al. found no significant indication that the common _FOXP2_ SNPs found in the general population are associated with population-level language ability differences.

# Exploratory Data Analysis Report

## Discussion: data quality issues
Though the data was freely and readily available, there was no explanation of what the data was or what it meant. An accompanyhing codebook would have helped navigate the ambiguity of raw scores vs. z-scores between the two datasets, as well as their notation for missing allele (0).

Presentation of the data in a more visual format instead of tables of text would have also helped better appreciate what the raw data was and what was being evaluated.

The authors also didn't discuss how exactly they curated the data for analysis. As we moved forward with replication, our counts did not match theirs, and we never did find the discrepancy.


## Stats and plots


Please Ensure that the foxp2_tidy before importing to ensure generation of tidy data. 
```{r read_in, message = FALSE}
elvs <- read_csv(here::here("data", "tidy_elvs.csv"))
elvs <- elvs %>% mutate(rs1916988 = as_factor(rs1916988))
iowa <- read_csv(here::here("data", "tidy_iowa.csv"))
iowa <- iowa %>% mutate_at(vars(starts_with("rs")), funs( as_factor(.)))
```
Since these the data comes in two seperate files, they will be assessed seperately. 

Additionally, since there is no easy way to keep factor levels stored in csv files, the recode statements are below:
```{r recode}
# C start
CC <- 'C/C'
CG <- 'C/G'
CT <- 'C/T'

#G start
GG <- 'G/G'
GA <- 'G/A'
GT <- 'G/T'

#T start
TT <-  'T/T'
TC <- 'T/C'


#A start
AA <- 'A/A'
AC <- 'A/C'
AG <- 'A/G'

#iowa %>% count(rs11505922) %>% glimpse() #used to formulate levels at each variant

# rs7791396
iowa  <-  iowa %>% mutate(rs7791396 = factor(rs7791396, levels = c(CC,CG,GG)))
# rs12155328
iowa <-  iowa %>% mutate(rs12155328 = factor(rs12155328, levels = c(CC,CT,TT)))
# rs10447760
iowa <-  iowa %>% mutate(rs10447760 = factor(rs10447760, levels = c(CC,CT,TT)))
# rs10953754
iowa <-  iowa %>% mutate(rs10953754 = factor(rs10953754, levels = c(AA,AG,GG)))
# rs2244419
iowa <-  iowa %>% mutate(rs2244419 = factor(rs2244419, levels = c(CC,CT,TT)))
# rs1668335
iowa <-  iowa %>% mutate(rs1668335 = factor(rs1668335, levels = c(AA,GA,GG)))
# rs2396720
iowa <-  iowa %>% mutate(rs2396720 = factor(rs2396720, levels = c(AA,AC,CC)))
# rrs1916988
iowa <-  iowa %>% mutate(rs1916988 = factor(rs1916988, levels = c(CC,TC,TT)))
# rs11505992
iowa <-  iowa %>% mutate(rs11505922 = factor(rs11505922, levels = c(CC,CT,TT)))
# rs7785701
iowa <-  iowa %>% mutate(rs7785701 = factor(rs7785701, levels = c(CC,CG,GG)))
# rs2106900
iowa <-  iowa %>% mutate(rs2106900 = factor(rs2106900, levels = c(AA,GA,GG)))
# rs7799652
iowa <-  iowa %>% mutate(rs7799652 = factor(rs7799652, levels = c(GG,GT,TT)))
# rs1005958
iowa <-  iowa %>% mutate(rs1005958 = factor(rs1005958, levels = c(CC,TC,TT)))


# ELVS dataset:
# rrs1916988
elvs <-  elvs %>% mutate(rs1916988 = factor(rs1916988, levels = c(CC,CT,TT)))

```

## ELVS dataset: 

###Raw data 
```{r elv_glimp}
elvs %>% glimpse()
```

number of variables: 3 (including the index we created (X1))  
number of observations: 308  
numerical variable: CELF7SS0 (the language score)  
categorical variable: rs1916988 (the SNP in the FOXP2 gene)

Each observation represents a research participant (a child around seven years of age).


###Compute summary statistics.
```{r}
elvs %>% skim()
```


A few things to note:  
1.  There are 308 observations, with no missing values.  (Originally there were 328 observations, but we removed 20 observations that had no genotype information.)  
2.  The three genotypes for the rs1916988 SNP are C/T, T/T, and C/C, with C/T being the most common (47%), closely followed by T/T (43%), and C/C being the least common (10%).  
3.  According to the paper, the language scores were standardized to have a mean of 100 and a standard deviation of 15.  The true mean and standard deviation calculated from this data are close to the stated values: 99.04 and 14.08, respectively.  
4.  The language scores range from 45 to 134.


### Distribution of the CELF7SS0 language scores

```{r celf}
elvs %>% ggplot(aes(x=CELF7SS0, fill = "red")) + geom_density() + geom_vline(xintercept = 100)
```

Here we can see that the distribution of the CELF7SS0 language score is centered at approximately 100. We also notice that it is left-skewed, indicating that we have a larger proportion of low scores present in this data set.  Therefore, it appears that children with weaker language skills were oversampled in this data set.  This study was conducted to find correlations between genotype and normal differences in language ability seen in the general population.  However, this higher prevalence of low scores suggests that this sample may not be entirely representative of the general population.

#### The frequency of each genotype for the rs1916988 SNP

```{r}
elvs %>% ggplot(aes(x=rs1916988,fill=rs1916988)) + geom_bar() 
```

Here we can see that the C/T genotype was the most common variant in the data set, with a frequency just a bit higher than the T/T genotype. Additionally there was a very small proportion of participants with the C/C genotype (just 10% of participants had this variant).


#### The distribution of the CELF7SS0 language scores, colored by genotype for rs1916988

```{r}
elvs %>% ggplot(aes(x=CELF7SS0, fill = rs1916988, alpha = 0.7)) + 
  geom_density() + 
  labs(title = "Language Scores from ELVS dataset, colored by Genotype variant for rs1916988")
```

This plot shows that the lower language scores are associated with the presence of the C allele.  This is consistent with what is reported in the article, though the authors state that the differences in the mean language scores for the three genotypes were not significant.  We can see that much of the data for all three variants does fall within a similar range of scores.

## IOWA dataset: 

###Raw data:

```{r}
iowa %>% glimpse()
```

number of variables: 17 (including the index we created (X1) and the ID that was assigned to each research participant)  
number of observations: 834
numerical variable: LCOMP (the language score)
categorical variables: Sample (the study from which the subject was sampled), each of 13 SNPs in the FOXP2 gene

Each observation represents a research participant (a kindergarten-age child from the Longitudinal study or an elementary school-age (first- through fourth-grade) child from the School study).

### Summary statistics:

```{r}
iowa %>% skim() 
```

A few things to note:  
1.  There are 834 observations, with no missing values.  However, all of the SNPs contain a significant proportion of entries labeled as "0/0."  It is presently unclear if these represent missing data or a homozygous deletion.  
2.  The three genotypes for the rs1916988 SNP (examined in the ELVS data set) are T/C, T/T, and C/C, with T/C being the most common (44%), followed by T/T (42%) and C/C being the least common (9%).  Possible deletion accounted for just 4%.
3.  The data was taken as a subset of data from two larger studies (Longitudinal and School).  The original samples of language scores were normalized to have a mean of 0 and a standard deviation of 1.  However, the mean for the selected subset of participants is -0.35, indicating that children with lower language ability were oversampled.  The standard deviation is slightly higher than that for the original data sets at 1.09.  
4.  The language scores range from -3.45 to 2.59.

#### Distribution of the LCOMP language scores, colored by the sample origin

```{r}
iowa %>% ggplot(aes(x=LCOMP,fill=Sample,alpha=.9)) + 
  geom_density()
```
The School sample is fairly symmetrical and centered approximately at 0, suggesting it is likely a good representation of the general population.  However, we can see that the Longitudinal sample is bimodal, with the taller peak (and therefore the majority of the language scores) below 0.  This is a clear illustration that the Longitudinal sample has an overabundance of children with lower-than-average language abilities (in spite of the authors' effort to correct for this bias when selecting their sample).  Thus, the Longitudinal sample does not appear to represent the general population.

#### The frequency of each genotype for the rs1916988 SNP

```{r}
iowa %>% ggplot(aes(x=rs1916988,fill = rs1916988, alpha = .9)) + 
  geom_bar(position = "stack") 
```
When comparing the same SNP in both the IOWA data set and the ELVS data set, we can see that there is a similar distribution of genotypes bewtween both samples. This indicates that at least both studies were designed similarly and sampled from similar populations.

```{r common_SNP}
common_SNP <- iowa %>% select(LCOMP, Sample, rs1916988)

common_SNP %>% 
  ggplot(aes(x = LCOMP, color = rs1916988, alpha = .7)) + geom_density() + facet_wrap(~Sample) + ggtitle('Iowa Score for rs1916988 by sample')
```

The reason the authors decided to look at the rs1916988 SNP in particular is because they got opposing results for this SNP when they looked at the Longitudinal and School samples separately.  The authors state that in the School sample, the T/T genotype was associated with significantly lower language scores than the T/C genotype.  In the Longitudinal sample, they got the opposite result, with the T/T genotype associated with higher language scores than either the T/C or C/C genotype, though this result was not significant.  From these plots, it is not apparent that these differences between genotypes would exist, as the distributions for either cohort seem to be quite similar for all genotypes.


# Replication Report:

## Table1:
"Means and standard deviations (SD) of language composite scores for tag SNP genotypes within FOXP2 and effect sizes($R^{2}$) of genotypes on language in the combined Iowa sample"
This table consists of summary statistics for each SNP in addition to the output from 13 one-way ANOVAs. 


To replicate this section we used the tidyverse package and base R to conduct the ANOVA on a subset of the data only containing the LCOMP scores and the specific SNP in question. First the mean, SD and counts are computed. Then an ANOVA is computed using the same subset data displaying the F statistic, R squared, and p value.


#### Observations: 

Here our analysis fully supports the authors' findings, there is no significant effect seen at any one SNP using ANOVA analysis. The unknown additional variables are still within the data set so the N's at each genotype are still slightly off. With that being said our observed statistics still closely match with the study and no large deviations are seen. 
The only slight deviation is rs12155328, which in the previous study approached significance but in this replication the statistic showed no signficance whatsoever.  


Function Declarations:
```{r tab_1_fun}
fun_mean <-  function(x){
  return(data.frame(y=mean(x),label=paste( 'Mean: ',formatC(mean(x,na.rm=T), digits =3, format='f'))))
}

fun_sd <-  function(x){
  return(data.frame(y=sd(x),label=paste( 'SD: ',formatC(sd(x,na.rm=T), digits =3, format='f'))))
}
```



#### rs7791396 ANOVA
```{r rs779}
print('rs7791396 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs7791396) %>% filter( !is.na(rs7791396) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs7791396) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs7791396 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs7791396,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs7791396))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```

This plot shows that there is no signficant difference found at rs7791396.
All statistics Mean and SD match for each genotype.
Here the n's for the C/G and G/G genotypes are off. 


#### rs12155328 ANOVA
```{r rs121}
print('rs12155328 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs12155328) %>% filter( !is.na(rs12155328) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs12155328) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs12155328 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs12155328,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs12155328))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs12155328.

Additionally this position does not approach signficiance as it does in the study.

All statistics Mean and SD match for each genotype.
Here the n's for the C/T and T/T genotypes are off. 

#### rs10447760 ANOVA
```{r rs104}
print('rs10447760 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs10447760) %>% filter( !is.na(rs10447760) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs10447760) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs10447760 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs10447760,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs10447760))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs10447760.

All statistics Mean and SD match for each genotype.
Here the n's for all genotypes are off. 

#### rs10953754 ANOVA
```{r rs109}
print('rs10953754 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs10953754) %>% filter( !is.na(rs10953754) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs10953754) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs10953754 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs10953754,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs10953754))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs10953754.
All statistics Mean and SD match for each genotype.
Here the n's for the A/G genotype are off. 


#### rs2244419 ANOVA
```{r rs224}
print('rs2244419 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs2244419) %>% filter( !is.na(rs2244419) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs2244419) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs2244419 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs2244419,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs2244419))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs2244419.
All statistics Mean and SD match for each genotype.
Here the n's for the A/G genotype are off. 

#### rs1668335 ANOVA
```{r rs166}
print('rs1668335 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs1668335) %>% filter( !is.na(rs1668335) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs1668335) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs1668335 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs1668335,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs1668335))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs1668335.
All statistics Mean and SD match for each genotype.
Here the n's for the A/G  and G/G genotypes are off. 

#### rs2396720 ANOVA
```{r rs239}
print('rs2396720 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs2396720) %>% filter( !is.na(rs2396720) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs2396720) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs2396720 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs2396720,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs2396720))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs2396720.
All statistics Mean and SD match for each genotype except for C/C mean which is -.05 smaller than the reported mean of -.40. This is due to the addition of extra data points!
Here the n's for the C/C genotypes are off. 

#### rs1916988 ANOVA
```{r rs191}
print('rs1916988 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs1916988) %>% filter( !is.na(rs1916988) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs1916988) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs1916988 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs1916988,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs1916988))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs1916988.
All statistics Mean and SD match for each genotype.
Here the n's for the C/T  and T/T genotypes are off. 

#### rs11505922 ANOVA
```{r rs115}
print('rs11505922 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs11505922) %>% filter( !is.na(rs11505922) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs11505922) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs11505922 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs11505922,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs11505922))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs11505922.
All statistics Mean and SD match for each genotype.
Here the n's for the C/T  and T/T genotypes are off. 


#### rs7785701 ANOVA
```{r rs778}
print('rs7785701 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs7785701) %>% filter( !is.na(rs7785701) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs7785701) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs7785701 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs7785701,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs7785701))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs7785701.
All statistics Mean and SD match for each genotype.
Here the n's for all genotypes are off. 

#### rs2106900 ANOVA
```{r rs210}
print('rs2106900 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs2106900) %>% filter( !is.na(rs2106900) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs2106900) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs2106900 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs2106900,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs2106900))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs2106900.
All statistics Mean and SD match for each genotype.
Here the n's for all genotypes are off.

#### rs7799652 ANOVA
```{r rs7799}
print('rs7799652 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs7799652) %>% filter( !is.na(rs7799652) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs7799652) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs7799652 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs7799652,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs7799652))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs7799652.
All statistics Mean and SD match for each genotype except mean of G/G.
Here the n's for all genotypes are off.

#### rs1005958 ANOVA
```{r rs100}
print('rs1005958 Statistics:')

#subset Data
test <- iowa %>% select(LCOMP,rs1005958) %>% filter( !is.na(rs1005958) , LCOMP != 0)

# Compute summary
sumar <- test %>% group_by(rs1005958) %>% summarize(
  mean(LCOMP),
  sd(LCOMP),
  n() 
  ) 

# Extract stats
mean <- sumar %>% pull('mean(LCOMP)')
sd <- sumar %>% pull('sd(LCOMP)')
n <- sumar %>% pull('n()')

# Display for each snp
(paste('Mean: ', mean, ' SD: ', sd, ' n: ', n))


# Create regression model
test_mod <- lm(LCOMP~ rs1005958 , data=test)

# Extract R-squared
rsq <- summary(test_mod)$r.squared[1]

(paste('R Squared: ' , rsq ))

# Extract F value
F_stat <- anova(test_mod)$'F value'[1]

(paste('F statistic: ' , F_stat))

# Extract p-value
p_val <- anova(test_mod)$'Pr(>F)'[1]

(paste('P value: ' , p_val))


#boxplot
test %>% ggplot(aes(x=rs1005958,y= LCOMP)) + 
  geom_boxplot(aes(fill=rs1005958))  +
  stat_summary(fun.data = fun_mean, geom='text',vjust=-1.5) +
  stat_summary(fun.data = fun_sd, geom='text',hjust=-.25) +
  theme_classic()
```
This plot shows that there is no signficant difference found at rs1005958.
All statistics Mean and SD match for each genotype.
Here the n's for C/T and T/T genotypes are off.



## Table2:

To examine rs1916988 more closely, the authors calculated the same statistics separately for the individual datasets in the IOWA sample (the Longitudinal and School samples in the table below).

![Table 2](Table2.png)


Below, we attempted to replicate their results.

##### IOWA Datasets

Our sample sizes for each genotype matched those reported for the Longitudinal dataset, with the exception that we had one additional participant for the C/C genotype.  All of our means matched as well.  Two of our standard deviations matched, but the third (for T/T) did not.

```{r stats_long}
# Longitudinal Sample
iowa_longitudinal <- iowa %>% filter(Sample == "Longitudinal")
iowa_longitudinal %>% group_by(rs1916988) %>% summarise(mean = mean(LCOMP), sd = sd(LCOMP), n = n())
```

For the School sample, our data included 21 additional observations that the authors did not include in their analysis, so our sample sizes were not matching up.  As we would expect, our mean and standard deviation values were getting thrown off as well.

```{r stats_school_1}
# School Sample
iowa_school <- iowa %>% filter(Sample == "School")
iowa_school %>% group_by(rs1916988) %>% summarise(mean = mean(LCOMP), sd = sd(LCOMP), n = n())
```

However, we noticed that there were exactly 21 observations in our data whose ID matched another ID, but with some additional numbers attached.  When we excluded these observations from our analysis, we were able to get the same sample sizes, means, and standard deviations that were reported in the paper.

```{r stats_school_2}
# School Sample
iowa_school <- iowa %>% filter(Sample == "School", !grepl("-", ID))
iowa_school %>% group_by(rs1916988) %>% summarise(mean = mean(LCOMP), sd = sd(LCOMP), n = n())
```

For the one-way ANOVAs, we first tried running them with the entire datasets.  However, this did not give us the R-squared values, F-statistics, and p-values reported in the paper.

```{r anova_iowa_1}
# Longitudinal Sample
iowa_longitudinal <- iowa %>% filter(Sample == "Longitudinal")
iowa_longitudinal_lm <- lm(LCOMP ~ rs1916988, iowa_longitudinal)
anova(iowa_longitudinal_lm)
summary(iowa_longitudinal_lm)$r.squared

# School Sample
iowa_school <- iowa %>% filter(Sample == "School")
iowa_school_lm <- lm(LCOMP ~ rs1916988, iowa_school)
anova(iowa_school_lm)
summary(iowa_school_lm)$r.squared
```

We suspected that LCOMP values of 0 were not legitimate values, so we decided to remove them.  While this didn't change anything for the Longitudinal dataset, this actually brought our values for the School sample farther from what the authors reported.

```{r anova_iowa_2}
# Longitudinal Sample
iowa_longitudinal <- iowa %>% filter(Sample == "Longitudinal", LCOMP != 0)
iowa_longitudinal_lm <- lm(LCOMP ~ rs1916988, iowa_longitudinal)
anova(iowa_longitudinal_lm)
summary(iowa_longitudinal_lm)$r.squared

# School Sample
iowa_school <- iowa %>% filter(Sample == "School", LCOMP != 0)
iowa_school_lm <- lm(LCOMP ~ rs1916988, iowa_school)
anova(iowa_school_lm)
summary(iowa_school_lm)$r.squared
```

We then tried removing just the genotypes labeled "0/0".  Originally, we thought that there was a chance that these might be homozygous deletions, but it really was not clear whether these were legitimate values.  Removing observations with the "0/0" genotype got us a lot closer to the reported values for both datasets.  For the Longitudinal dataset, we essentially matched the reported values, with an R-squared value within 0.0002 of the reported value, a perfectly matched F-statistic, and a p-value within 0.01.

```{r anova_iowa_3}
# Longitudinal Sample
iowa_longitudinal <- iowa %>% filter(Sample == "Longitudinal", rs1916988 != "0/0")
iowa_longitudinal_lm <- lm(LCOMP ~ rs1916988, iowa_longitudinal)
anova(iowa_longitudinal_lm)
summary(iowa_longitudinal_lm)$r.squared

# School Sample
iowa_school <- iowa %>% filter(Sample == "School", rs1916988 != "0/0")
iowa_school_lm <- lm(LCOMP ~ rs1916988, iowa_school)
anova(iowa_school_lm)
summary(iowa_school_lm)$r.squared
```

While the School values were a lot closer, they still were not matching the values the authors reported.  The F-statistic was just a bit too high, and the p-value was just a little too low.  When we had removed the language scores with "0" values, the F-statistic had moved slightly down, and the p-value moved slightly up.  Therefore, we tried eliminating any entries that had either a "0/0" genotype or a language score of "0".  These filters resulted in a matching R-squared and p-value, and an F-statistic that was within 0.01 of the reported value.

```{r anova_iowa_4}
# School Sample
iowa_school <- iowa %>% filter(Sample == "School",
                               LCOMP != 0, rs1916988 != "0/0")
iowa_school_lm <- lm(LCOMP ~ rs1916988, iowa_school)
anova(iowa_school_lm)
summary(iowa_school_lm)$r.squared
```

When we realized that the authors had eliminated 21 observations with duplicate IDs from the School dataset, we noticed that many (though not all) of these observations had LCOMP scores of 0.  We decided to test whether we could still get the reported values by filtering to remove these observations in lieu of all 0-value LCOMP scores.  What we discovered was that we not only kept our matching R-squared and p-values, but we also got our F-statistic to match perfectly as well.  Therefore, we suspect that this is how the authors filtered the School dataset.

```{r anova_iowa_5}
# School Sample
iowa_school <- iowa %>% filter(Sample == "School",
                               !grepl("-", ID), rs1916988 != "0/0")
iowa_school_lm <- lm(LCOMP ~ rs1916988, iowa_school)
anova(iowa_school_lm)
summary(iowa_school_lm)$r.squared
```

The following tables summarize our results for the Longitudinal and School samples.

**Longitudinal Sample**

```{r table_long}
iowa_longitudinal %>% group_by(rs1916988) %>% summarise(mean = mean(LCOMP), sd = sd(LCOMP), n = n())

longitudinal_table <- data.frame("R-squared" = 0.0080, "F-statistic" = 1.89, "p-value" = 0.15)
longitudinal_table
```

**School Sample**

```{r table_school}
iowa_school %>% group_by(rs1916988) %>% summarise(mean = mean(LCOMP), sd = sd(LCOMP), n = n())

school_table <- data.frame("R-squared" = 0.03, "F-statistic" = 4.24, "p-value" = 0.015)
school_table
```

In agreement with the authors' findings, we see that in the Longitudinal sample, the C allele is associated with a lower language score than the T allele, but in the School sample, the reverse is true.  Only the result from the School sample is nominally statistically significant, although the p-value should really be adjusted to account for the fact that this study involved multiple comparisons.


##### ELVS Dataset

Based on their contradictory findings for rs1916988, the authors performed the same statistical analysis for the SNP in the ELVS sample.  To try to match the mean and standard deviation of the ELVS data that were reported in the paper, we had to normalize the data.

```{r normalize_elvs_1}
elvs_normalized <- elvs %>% standardize()
head(elvs_normalized)
```

We had previously removed 20 observations that had no genotype information, and we performed our statistical analysis on the remaining data.  Our sample sizes for each genotype match those in the paper, so we didn't feel that we needed to add or remove any data.  However, our mean and standard deviation values do not match those in the paper.

```{r stats_elvs_1}
elvs_normalized %>% group_by(rs1916988) %>% summarise(mean = mean(CELF7SS0), sd = sd(CELF7SS0), n = n())
```

We tested the data to be sure they had been normalized to have a mean of 0 and a standard deviation of 1.

```{r normalize_check}
elvs_normalized %>% summarise(mean = mean(CELF7SS0), sd = sd(CELF7SS0), n = n())
```


We see that the mean is virtually 0, and the standard deviation is 1.  Just to be sure the normalization method was not causing some sort of issue, we decided to try another normalization method.

```{r normalize_elvs_2}
elvs_normalized <- normalize(elvs, method = "standardize")
head(elvs_normalized)
```

However, even with a different normalization nethod, we still got the same results.

```{r anova_elvs_2}
elvs_normalized %>% group_by(rs1916988) %>% summarise(mean = mean(CELF7SS0), sd = sd(CELF7SS0), n = n())
```


When we checked whether our R-squared, F-statistic, and p-value matched the authors' data, our R-squared value was only 0.0002 away from the reported value, and our F-statistic and p-value matched perfectly.

```{r anova_elvs}
elvs_lm <- lm(CELF7SS0 ~ rs1916988, elvs_normalized)
anova(elvs_lm)
summary(elvs_lm)$r.squared
```

The following tables summarize our results for the ELVS sample.

```{r elvs_table}
elvs_normalized %>% group_by(rs1916988) %>% summarise(mean = mean(CELF7SS0), sd = sd(CELF7SS0), n = n())

elvs_table <- data.frame("R-squared" = 0.0082, "F-statistic" = 1.23, "p-value" = 0.29)
elvs_table
```

Similar to what we saw in the Longitudinal sample, the C allele appears to be associated with lower language scores than the T allele. However, this result is still not significant.  Therefore, the ELVS sample does not suggest any significant effect of variation at this SNP on language ability.  This is, indeed, what the authors had concluded.


## Table4:
"SNP location, minor allele frequency and call rates for markers in FOXP2"
Here the study table shows extreneous information such as SNP location and minor allele frequency.

For the following analysis only the SNP call rate is computed to prove there is an acceptable call rate as quality control for accurate genotype calls.
replicate call rates for iowa dataset 


Subsetting the data can making variables to be called.
```{r}
tot <- iowa %>% count() %>% pull(n)

#subset long
long <- iowa %>% filter(Sample == 'Longitudinal')
l_tot <- long %>% count() %>% pull(n)

#Subset school
sch <- iowa %>% filter(Sample == 'School')
s_tot <- sch %>% count() %>% pull(n)

```


##rs7791396
```{r}

t_call <- iowa %>% filter(!is.na(rs7791396)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs7791396)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs7791396)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings! 

##rs12155328
```{r}

t_call <- iowa %>% filter(!is.na(rs12155328)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs12155328)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs12155328)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.

##rs10447760
```{r}

t_call <- iowa %>% filter(!is.na(rs10447760)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs10447760)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs10447760)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.

##rs10953754
```{r}

t_call <- iowa %>% filter(!is.na(rs10953754)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs10953754)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs10953754)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.


##rs2244419
```{r}

t_call <- iowa %>% filter(!is.na(rs2244419)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs2244419)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs2244419)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.


##rs1668335
```{r}

t_call <- iowa %>% filter(!is.na(rs1668335)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs1668335)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs1668335)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.

##rs2396720
```{r}

t_call <- iowa %>% filter(!is.na(rs2396720)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs2396720)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs2396720)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.

##rs1916988
```{r}

t_call <- iowa %>% filter(!is.na(rs1916988)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs1916988)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs1916988)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.

##rs11505922
```{r}

t_call <- iowa %>% filter(!is.na(rs11505922)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs11505922)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs11505922)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.

##rs7785701
```{r}

t_call <- iowa %>% filter(!is.na(rs7785701)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs7785701)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs7785701)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Longitudinal is off by .01.


##rs2106900
```{r}

t_call <- iowa %>% filter(!is.na(rs2106900)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs2106900)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs2106900)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.

##rs7799652
```{r}

t_call <- iowa%>% filter(!is.na(rs7799652)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs7799652)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs7799652)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.

##rs1005958
```{r}

t_call <- iowa %>% filter(!is.na(rs1005958)) %>% count()
t_call <- signif(t_call/tot, 3)
l_call <- long %>% filter(!is.na(rs1005958)) %>% count()
l_call <- signif(l_call/l_tot,3)
s_call <- sch %>% filter(!is.na(rs1005958)) %>% count()
s_call <- signif(s_call/s_tot,3)

calls <- tibble('long' = c(l_call), 'school' =c(s_call),'comb'= c(t_call))

calls %>% glimpse()
```

Matches Previous Findings.



# Extension Report:

## Question:

  Based on the biology of gene function, it is possible that these FOXP2 variants show a combinatorial effect of language score. This is expected as the a small change in gene struture at one position may not be effect to alter function and thus the observed phenotype. Additionally, not all SNPs are created equal, and combination dependent require two (or more) specific SNPs. 
 
## ANCOVA:
With levels set now we can start to conduct a ANCOVA looking for snps that interact through brute force checking each combination of SNPs to see what is interacting. I will also only conduct one non significant analysis and then copy and paste each. 

Note: Interaction model is the same regardless of which order (type II) So I will skip repeat interactions. 

Of interest is an interaction in order to validate an ANCOVA.


All the SNPs present: 
rs7791396, rs12155328, rs10447760, rs10953754, rs2244419, rs1668335, rs2396720, rs1916988, rs11505922, rs7785701, rs2106900, rs7799652, rs1005958


# rs7791396 v rs12155328: Not Significant
```{r rs779vrs121}
mod <- lm(LCOMP~rs7791396*rs12155328, data= iowa, contrasts = list(rs7791396 = contr.sum, rs12155328 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```

No interactions present.



##rs7791396 V rs1005958: Significant

```{r rs779_v_rs100}
mod <- lm(LCOMP~rs7791396*rs1005958, data= iowa, contrasts = list(rs7791396 = contr.sum, rs1005958 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```


Significance found between rs7791396 and rs1005958!! Very surprsing indeed.
Unable to conduct ANCOVA between rs7791396 and rs1005958. 
ANCOVA assumes no interactions. Can do interaction effect analysis to see which Genotypes give the most change. 


##rs12155328 V rs10953754: Not Signfiant enough
```{r rs121_v_rs109}
mod <- lm(LCOMP~rs12155328*rs10953754, data= iowa, contrasts = list(rs12155328 = contr.sum, rs10953754 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```


.1 level of significance my scientific instincts tell me to report this. 
The interaction isnt signficant but it appears that rs12155328 begins to predict LCOMP when interacting with rs10953754 in a vacuum .

Will still be able to conduct ANCOVA.




Additonally happens marginal signficance  (> .1) happens at:

  * rs12155328 v rs1668335
  * rs12155328 v rs1916988
  * rs12155328 v rs7785701
  * rs12155328 v rs2106900
  * rs12155328 v rs7799652
  


##rs10447760 V rs11505922: Signifiant 
```{r rs104_v_rs115}
mod <- lm(LCOMP~rs10447760*rs11505922, data= iowa, contrasts = list(rs10447760 = contr.sum, rs11505922 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```

Signficant Interaction found!!!
Unable to conduct ANCOVA between rs10447760 and rs11505922.
ANCOVA assumes no interactions. Can do spotlight analysis to see which Genotypes give the most change. 


##rs10447760 V rs7785701: Weakly Significant 
```{r rs104_v_rs778}
mod <- lm(LCOMP~rs10447760*rs7785701, data= iowa, contrasts = list(rs10447760 = contr.sum, rs7785701 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```

Weakly Signficant Interaction found.
Unable to conduct ANCOVA between rs10447760 and rs7785701.
ANCOVA assumes no interactions. Can do spotlight analysis to see which Genotypes give the most change. 


##rs10447760 V rs2106900: Weakly Signfiant 
```{r rs104_v_rs210}
mod <- lm(LCOMP~rs10447760*rs2106900, data= iowa, contrasts = list(rs10447760 = contr.sum, rs2106900 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```
Weakly Signficant Interaction found.
Unable to conduct ANCOVA between rs10447760 and rs2106900.
ANCOVA assumes no interactions. Can do spotlight analysis to see which Genotypes give the most change. 


##rs2244419 V rs1668335: Signfiant 
```{r rs224_v_rs166}
mod <- lm(LCOMP~rs2244419*rs1668335, data= iowa, contrasts = list(rs2244419 = contr.sum, rs1668335 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```
Signficant Interaction found.
Unable to conduct ANCOVA between rs2244419 and rs1668335. 
ANCOVA assumes no interactions. Can do spotlight analysis to see which Genotypes give the most change.

##rs2244419 V rs1916988: Signfiant 
```{r rs224_v_rs191}
mod <- lm(LCOMP~rs2244419*rs1916988, data= iowa, contrasts = list(rs2244419 = contr.sum, rs1916988 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```
Signficant Interaction found.
Unable to conduct ANCOVA between rs2244419 and rs1916988.
ANCOVA assumes no interactions. Can do spotlight analysis to see which Genotypes give the most change.


##rs1668335 V rs11505922: Weakly Signfiant 
```{r rs166_v_rs115}
mod <- lm(LCOMP~rs1668335*rs11505922, data= iowa, contrasts = list(rs1668335 = contr.sum, rs11505922 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```
Weakly Signficant Interaction found.
Unable to conduct ANCOVA between rs10447760 and rs2106900. 
ANCOVA assumes no interactions. Can do spotlight analysis to see which Genotypes give the most change.

##rs1668335 V rs2106900: Weakly Signfiant 
```{r rs166_v_rs210}
mod <- lm(LCOMP~rs1668335*rs2106900, data= iowa, contrasts = list(rs1668335 = contr.sum, rs2106900 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```
Weakly Signficant Interaction found.
Unable to conduct ANCOVA between rs10447760 and rs2106900.
ANCOVA assumes no interactions. Can do spotlight analysis to see which Genotypes give the most change.

##rs1916988 V rs11505922: Signfiant 
```{r rs191_v_rs115}
mod <- lm(LCOMP~rs1916988*rs11505922, data= iowa, contrasts = list(rs1916988 = contr.sum, rs11505922 = contr.sum))

Anova(mod, type=2, singular.ok = T)

```
Signficant Interaction found!!!
Unable to conduct ANCOVA between rs10447760 and rs11505922.
ANCOVA assumes no interactions. Can do spotlight analysis to see which Genotypes give the most change. 




End of Interaction Analysis. 

#Recap: 

Significant Interactions Found: 

  Will conduct spotlight analysis on most signficant interaction.
  
  * rs7791396 V rs1005958 p-value = 0.03221

  * rs10447760 v rs11505922 p-value = 0.004627
  
  * rs2244419 V rs1668335 p-value = 0.01351
  
  * rs2244419 V rs1916988 p-value = 0.02436
  
  * rs1916988 V rs11505922 p-value = 0.03526
  
  
  This is the only variant to reach < .05 level of significance and as such the spotlight analysis will only be conducted on this variant. 
  
  
  Weak covariate: rs12155328
  
  This was the only variant to display this covaraince. 
  

# Spotlight Analysis:

Now conducting spotlight analysis on rs10447760 against rs11505922.


For categorical variables an interaction effects test is the same as spotlight analysis.

```{r ancov104}
#Subsetting data
sig_mod <- lm(LCOMP~rs10447760*rs11505922, data= iowa, contrasts = list(rs10447760 = contr.sum, rs11505922 = contr.sum))

#ANOVA output for verification
#Anova(sig_mod, type=2, singular.ok = T)

#Reverse each to see
testInteractions(sig_mod,fixed='rs10447760',across='rs11505922', adjustment= 'bonferroni')

testInteractions(sig_mod,fixed='rs11505922',across='rs10447760', adjustment= 'bonferroni')
```



Significant interaction between Genotype 'C/C' in rs11505922 and T/T in rs10447760.




```{r ancov779}
#Subsetting data
sig_mod <- lm(LCOMP~rs7791396*rs1005958, data= iowa, contrasts = list(rs7791396 = contr.sum, rs1005958 = contr.sum))

#ANOVA output for verification
#Anova(sig_mod, type=2, singular.ok = T)

#Reverse each to see
testInteractions(sig_mod,fixed='rs7791396',across='rs1005958', adjustment= 'bonferroni')

testInteractions(sig_mod,fixed='rs1005958',across='rs7791396', adjustment= 'bonferroni')
```

Here it is rs7791396 correlates G/G with rs1005958 T/C.


```{r ancov224}
#Subsetting data
sig_mod <- lm(LCOMP~rs2244419*rs1668335, data= iowa, contrasts = list(rs2244419 = contr.sum, rs1668335 = contr.sum))

#ANOVA output for verification
#Anova(sig_mod, type=2, singular.ok = T)

#Reverse each to see
#testInteractions(sig_mod,fixed='rs2244419',across='rs1668335', adjustment= 'bonferroni')

#testInteractions(sig_mod,fixed='rs1668335',across='rs2244419', adjustment= 'bonferroni')
```

This Analysis wont work because they use the same genotype indices.
Error in linearHypothesis.lm(model, L, ...) : there are aliased coefficients in the model.


```{r ancov224_166 }
#Subsetting data
sig_mod <- lm(LCOMP~rs2244419*rs1668335, data= iowa, contrasts = list(rs2244419 = contr.sum, rs1668335 = contr.sum))

#ANOVA output for verification
#Anova(sig_mod, type=2, singular.ok = T)

#Reverse each to see
#testInteractions(sig_mod,fixed='rs2244419',across='rs1668335', adjustment= 'bonferroni')

#testInteractions(sig_mod,fixed='rs1668335',across='rs2244419', adjustment= 'bonferroni')
```

This Analysis wont work because they use the same genotype indices.
Error in linearHypothesis.lm(model, L, ...) : there are aliased coefficients in the model.


```{r ancov191}
#Subsetting data
sig_mod <- lm(LCOMP~rs1916988*rs11505922, data= iowa, contrasts = list(rs1916988 = contr.sum, rs11505922 = contr.sum))

#ANOVA output for verification
Anova(sig_mod, type=2, singular.ok = T)

#Reverse each to see
testInteractions(sig_mod,fixed='rs1916988',across='rs11505922', adjustment= 'bonferroni')

testInteractions(sig_mod,fixed='rs11505922',across='rs1916988', adjustment= 'bonferroni')
```

No Significance found at this position.


## Anova Summary:

Based on this analysis there were some SNP combinations that showed some signfiance revealing small interactions between certain genotypes. This makes sense as only certain genotype combinations start to affect gene function. But please note, for this analysis there were 73 total comparisons, so if these values were p-value adjusted none of these findings would be significant. 

The interaction of these SNPs were also investigated by the study using a random forest machine learning algorithm revealing that there are no signficiant SNPs or interactions that directly affect LCOMP score. 

